main.c

```diff
@@ -1,173 +1,66 @@
-                }
-                state++;
-                drinkChoice=choice-'a';
+            case 3:{
+                displayDrinkOptions(noOfDrinks,food[foodChoice],drinks,priceDrinks);
+                drinkChoice = getChoiceIndex(noOfDrinks, &state);
                 break;
             }
-            case 4:
-            {
-                //Cutlery
-                printf("Do you want cutlery?\n");
-                printf("a) Yes\n");
-                printf("b) No, thanks!\n");
-                printf("c) Go back\n");
-                choice=getchar();
-                getchar();
-                if(choice=='c')
-                {
-                    state--;
-                    break;
-                }
-                else
-                    if(choice=='a')
-                        cutlery=1;
-                    else
-                        cutlery=0;
-                state++;
+            case 4:{
+                displayCutlery();
+                cutlery=getChoiceIndex(2,&state)+1;
                 break;
             }
-            case 5:
-            {
-                //Additional info
-                printf("Any additional info?\n");
-                gets(addInfo);
+            case 5:{
+                addAdditionalInformation(addInfo);
             }
-            case 6:
-            {
-                //Display Order
-                printf("This is your order:\n");
-                printf("-------------------\n");
-                printf("Name: %s\n", username);
-                printf("Food items:\n");
-                printf("---%s: (%0.2f)\n",type[foodChoice][typeChoice],prices[foodChoice][typeChoice]);
-                if(drinkChoice!=noOfDrinks) {
-                    printf("---%s: (%0.2f)\n", drinks[drinkChoice], priceDrinks[drinkChoice]);
-                }
-                if(cutlery==1) {
-                    printf("Cutlery: ");
-                    printf("yes\n");
-                }
-               // else
-                    //printf("no\n");
-                if(strcmp(addInfo,"")!=0) {
-                    printf("Additional info: %s\n", addInfo);
-                }
-                printf("Payment amount (%0.2f)\n",prices[foodChoice][typeChoice]+priceDrinks[drinkChoice]);
-                printf("--------------------\n");
-                printf("a) Confirm order\n");
-                printf("b) Go back\n");
-                choice = getchar();
-                if(choice=='a') {
-                    confirmed = 1;
-                    printf("Order confirmed! Thank you for buying from us, %s!\n",username);
-                }
-                else{
-                    state--;
-                }
-                getchar();
+            case 6:{
+                displayFoodData(type[foodChoice][typeChoice],prices[foodChoice][typeChoice],drinks[drinkChoice], priceDrinks[drinkChoice], username);
+                showCutlery(cutlery);
+                showAddInfo(addInfo);
```

Comment:
could be moved to a displayOrder function (keep these 3 functions, just call them from a new one)